### PR TITLE
Fix strict template issues for code editor

### PR DESCRIPTION
**Pull request by ivanchimeno**
### Motivation and Context

This pr fixes angular strict template issues for `/app/exercises/programming/shared/code-editor` and `/app/exercises/programming/manage/programming-exercise-detail.component.html`

### CODE REVIEW

**Review by SimonEntholzer — APPROVED** [review]
Looks good, couldn't find any issues

File: src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.html [hunk 1]
```diff
@@ -235,23 +235,23 @@ <h2><span jhiTranslate="artemisApp.programmingExercise.detail.title">Programming
...
+                        ><a jhiBuildPlanLink [buildPlanId]="programmingExercise.templateParticipation.buildPlanId" [projectKey]="programmingExercise.projectKey!">{{
                             programmingExercise.templateParticipation.buildPlanId
                         }}</a></span
                     >
                 </dd>
                 <dt><span jhiTranslate="artemisApp.programmingExercise.solutionBuildPlanId">Solution Build Plan Id</span></dt>
                 <dd>
                     <span *ngIf="programmingExercise.solutionParticipation && programmingExercise.solutionParticipation.buildPlanId"
-                        ><a jhiBuildPlanLink [buildPlanId]="programmingExercise.solutionParticipation.buildPlanId" [projectKey]="programmingExercise.projectKey">{{
+                        ><a jhiBuildPlanLink [buildPlanId]="programmingExercise.solutionParticipation.buildPlanId!" [projectKey]="programmingExercise.projectKey!">{{
```
Comment (SimonEntholzer) [inline src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.html:254]: any reason we need a `!` after buildplan here but not above in the templateParticipation? For both it is defined as:
`buildPlanId?: string`

Comment (ivanchimeno) [reply src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.html:254]: Good question. I'll remove it and see if the build fails

Comment (ivanchimeno) [reply src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.html]: Indeed it doesn't cause a problem so I reverted that change.

**Review by SimonEntholzer — APPROVED** [review]
Re-approve, recent changes are fine

**Review by jamalxo — APPROVED** [review]
looks good to me

**Review by imstreetspirit — APPROVED** [review]
tested on ts1, find nothing. fixes seems good.

**Pull request merged**